src/Facets/AllBridgeFacet.sol

```diff
@@ -0,0 +1,148 @@
+
+/// @title Allbridge Facet
+/// @author Li.Finance (https://li.finance)
+/// @notice Provides functionality for bridging through AllBridge
+contract AllBridgeFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// @notice The contract address of the AllBridge router on the source chain.
+    IAllBridge public immutable allBridge;
+
+    /// @notice The struct for the AllBridge data.
+    /// @param fees The amount of token to pay the messenger and the bridge
+    /// @param recipient The address of the token receiver after bridging.
+    /// @param destinationChainId The destination chain id.
+    /// @param receiveToken The token to receive on the destination chain.
+    /// @param nonce A random nonce to associate with the tx.
+    /// @param messenger The messenger protocol enum
+    struct AllBridgeData {
+        uint256 fees;
+        bytes32 recipient;
+        uint8 destinationChainId;
+        bytes32 receiveToken;
+        uint256 nonce;
+        MessengerProtocol messenger;
+    }
+
+    /// @notice Initializes the AllBridge contract
+    /// @param _allBridge The address of the AllBridge contract
+    constructor(IAllBridge _allBridge) {
+        allBridge = _allBridge;
+    }
+
+    /// @notice Bridge tokens to another chain via AllBridge
+    /// @param _bridgeData The bridge data struct
+    function startBridgeTokensViaAllBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        AllBridgeData calldata _allBridgeData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _allBridgeData);
+    }
+
+    /// @notice Bridge tokens to another chain via AllBridge
+    /// @param _bridgeData The bridge data struct
+    /// @param _swapData The swap data struct
+    /// @param _allBridgeData The AllBridge data struct
+    function swapAndStartBridgeTokensViaAllBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        AllBridgeData calldata _allBridgeData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _allBridgeData);
+    }
+
+    /// @notice converts an address to bytes
+    /// @param a The address to convert
+    function _toBytes(address a) internal pure returns (bytes memory) {
```

Comment:
Is this function still used anywhere? Seems outdated and could be removed

Reply:
Removed